container/go/pkg/oci/reader_test.go

```diff
@@ -0,0 +1,123 @@
+package oci
+
+import (
+	"testing"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+	"github.com/google/go-containerregistry/pkg/v1/validate"
+)
+
+var readertests = []struct {
+	// A descriptive name for this test case.
+	name string
+	// The hash code of the manifest metadata file.
+	manifestDigest v1.Hash
+	// The hash code of the config metadata file.
+	configDigest v1.Hash
+	// An array of hash codes for each layer in this image.
+	layerHashes []v1.Hash
+	// The media type of this image.
+	mediaType types.MediaType
+	// The relative path of this test case.
+	testPath string
+}{
+	// #1 This test index is the output of puller.go
```

Comment:
As a reader of a test case, I'm also interested in a tl; dr of what exactly this test case is trying to do. So comment something like:
`// Ensure the image downloaded by puller.go from <image path> located in testdata/test_index1 can be loaded.`